src/raiden_libs/blockchain.py

```diff
@@ -119,30 +120,40 @@ def parse_token_network_event(event: dict) -> Optional[Event]:
 
     common_infos = dict(
         token_network_address=decode_hex(event["address"]),
-        channel_identifier=event["args"]["channel_identifier"],
         block_number=event["blockNumber"],
     )
 
     if event_name == ChannelEvent.OPENED:
         return ReceiveChannelOpenedEvent(
+            channel_identifier=event["args"]["channel_identifier"],
             participant1=to_canonical_address(event["args"]["participant1"]),
             participant2=to_canonical_address(event["args"]["participant2"]),
             settle_timeout=event["args"]["settle_timeout"],
             **common_infos,
         )
     if event_name == ChannelEvent.CLOSED:
         return ReceiveChannelClosedEvent(
+            channel_identifier=event["args"]["channel_identifier"],
             closing_participant=to_canonical_address(event["args"]["closing_participant"]),
             **common_infos,
         )
     if event_name == ChannelEvent.BALANCE_PROOF_UPDATED:
         return ReceiveNonClosingBalanceProofUpdatedEvent(
+            channel_identifier=event["args"]["channel_identifier"],
             closing_participant=to_canonical_address(event["args"]["closing_participant"]),
             nonce=event["args"]["nonce"],
             **common_infos,
         )
     if event_name == ChannelEvent.SETTLED:
-        return ReceiveChannelSettledEvent(**common_infos)
+        return ReceiveChannelSettledEvent(
+            channel_identifier=event["args"]["channel_identifier"],
+            **common_infos,
+        )
+
+    # FIXME: use value from ChannelEvent as soon as PR is merged
+    # https://github.com/raiden-network/raiden-contracts/pull/1389
+    if event_name == "DeprecationSwitch":
+        return DeprecationSwitchEvent(**common_infos)
```

Comment:
We can even argue if we should add the event as long as we don't use it to trigger a certain action.